src/main/java/com/yahoo/memory/CompareAndCopy.java

```diff
@@ -132,35 +149,30 @@ private static boolean equalsByBytes(final Object arr1, final long cumOff1, fina
    * <a href="https://github.com/OpenHFT/Zero-Allocation-Hashing/blob/master/src/main/java/net/openhft/hashing/XxHash.java">
    * OpenHFT/Zero-Allocation-Hashing</a>, which has an Apache 2 license as does this site.
    */
-  static int hashCode(final BaseState state) {
-    state.checkValid();
-    final long lengthBytes = state.getCapacity();
-    long offsetBytes = state.getCumulativeOffset();
-    final Object arr = state.getUnsafeObject(); //could be null
-
+  private static long xxHash64(final Object unsafeObj, long offsetBytes, final long lengthBytes) {
```

Comment:
This is very dangerous misuse of parameter naming.  Throughout the rest of the Memory implementation, "offsetBytes" is a user-specified offset from the user's concept of the start of data for a Memory (or Buffer) object.   What is really required here is "cumOffsetBytes", which including any user given offsetBytes and may also include other offset components such as the native off-heap memory address, DirectByteBuffer split offsets, region offsets, and unsafe arrayBaseOffsets.  

Having hashing methods here also unnecessarily clutters this class, which is already more than just CompareAndCopy. 

To correct this I decided to put the hashing into its own class with clear documentation of the parameters.

Fixed.